yt/frontends/amrvac/fields.py

```diff
@@ -40,10 +41,10 @@ def _velocity(field, data, idir, prefix=None):
 
     mask1 = rho == 0
     if mask1.any():
-        mylog.info(
-            "zeros found in %sdensity, "
+        warnings.warn(
```

Comment:
So, is there a reason this is being switched over to a warning? My impression is that warnings.warn should be used if the user can do something about it, and that logging.warning should be used if the event should be noted but isn't something that can be changed by the user?  

The table [in the logging docs](https://docs.python.org/3/howto/logging.html#when-to-use-logging) about issuing warnings distinguishes these events.

Reply:
I see the point in the PR description about the copious printing, but I wonder if there is a solution consistent with the use of logger/warning

Reply:
That's an excellent point I didn't think of. For reference, here's an issue to discuss it more thoroughly https://github.com/yt-project/yt/issues/2865